data-products/src/main/scala/org/sunbird/analytics/archival/BaseArchivalJob.scala

```diff
@@ -122,29 +126,48 @@ trait BaseArchivalJob extends BaseReportsJob with IJob with ArchivalMetaDataStor
       .withColumn("question", to_json(col("question")))
   }
 
-  def archiveBatches(batchesToArchive: Map[String, Array[BatchPartition]], data: DataFrame, archivalRequest: ArchivalRequest, request: Request)(implicit config: JobConfig): Unit = {
-    batchesToArchive.flatMap(batches => {
+  def archiveBatches(batchesToArchive: Map[String, Array[BatchPartition]], data: DataFrame, requestConfig: Request)(implicit config: JobConfig): Unit = {
+    batchesToArchive.foreach(batches => {
       val processingBatch = new AtomicInteger(batches._2.length)
       JobLogger.log(s"Started Processing to archive the data", Some(Map("batch_id" -> batches._1, "total_part_files_to_archive" -> processingBatch)))
+
       // Loop through the week_num & year batch partition
-      val res = for (batch <- batches._2.asInstanceOf[Array[BatchPartition]]) yield {
+      batches._2.map((batch: BatchPartition) => {
```

Comment:
Why are we keeping the archival logic in base class ? Please implement in the assessment archival job.
let subclass to override the archival logic. Similar to how we are doing in the exhaust job of process method.